internal/address/gcp.go

```diff
@@ -214,20 +214,23 @@ func (a *gcpAssigner) CheckAddressAssigned(region, addressName string) (bool, er
 	return address.Status == inUseStatus, nil
 }
 
-func (a *gcpAssigner) Assign(ctx context.Context, instanceID, zone string, filter []string, orderBy string) error {
+func (a *gcpAssigner) Assign(ctx context.Context, instanceID, zone string, filter []string, orderBy string) (string, error) {
 	// check if instance already has a public static IP address assigned
-	instance, err := a.checkStaticIPAssigned(zone, instanceID)
+	instance, address, err := a.checkStaticIPAssigned(zone, instanceID)
 	if err != nil {
-		return errors.Wrapf(err, "check if static public IP is already assigned to instance %s", instanceID)
+		if errors.Is(err, ErrStaticIPAlreadyAssigned) {
```

Comment:
Is this a change in behaviour? If so, should it go in a separate PR?

Reply:
Sort of, the `ErrStaticIPAlreadyAssigned` error was being caught by the caller anyway, but we need a way to return the assigned static address so we can wait for it to be reported later.